modules/collect/src/test/java/com/opengamma/strata/collect/result/PartialResultTest.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.collect.result;
+
+import static com.opengamma.strata.collect.TestHelper.assertSerialization;
+import static com.opengamma.strata.collect.TestHelper.coverBeanEquals;
+import static com.opengamma.strata.collect.TestHelper.coverImmutableBean;
+import static org.testng.Assert.assertEquals;
+
+import org.testng.annotations.Test;
+
+import com.google.common.collect.ImmutableList;
+
+/**
+ * Test {@link PartialResult}.
+ */
+@Test
+public class PartialResultTest {
```

Comment:
This design matches exactly what I've been using in a client project. The value has to have a notion of empty. In most cases it's an empty list or an empty map, but in some cases it's a custom data structure which can be empty. It's up to the caller to decide whether it's successful or not.